realm/realm-library/src/androidTest/java/io/realm/RealmTests.java

```diff
@@ -1567,6 +1572,114 @@ public void copyToRealmOrUpdate_boxedLongPrimaryKeyFieldIsNull() {
         assertEquals(SECONDARY_FIELD_UPDATED, realm.allObjects(PrimaryKeyAsBoxedLong.class).first().getName());
     }
 
+    // @PrimaryKey + @Required annotation accept not-null value properly as a primary key value for Realm version 0.89.1+
+    @Test
+    public void copyToRealmOrUpdate_primaryKeyRequiredFieldItNotNull() {
```

Comment:
I would rename this to `copyToRealmOrUpdate_requiredPrimaryKey` . NotNull is implied in required, so no need to mention it twice IMO